roles/system_service/tasks/main.yml

```diff
@@ -3,16 +3,14 @@
   template:
     src: etc/systemd/system/service.j2
     dest: "/etc/systemd/system/{{ service_name }}.service"
-  when: systemd_enabled == true
+  notify: Restart service
```

Comment:
handlers are executed at the end of the play so the daemon will be reloaded first.